src/test/java/hudson/plugins/audit_trail/LogFileAuditLoggerTest.java

```diff
@@ -36,7 +48,131 @@ public void environmentVariablesAreProperlyExpanded() {
+        Assert.assertEquals(directoryFiles.size(), 1);
+
+        String log = Util.loadFile(logFileRotating.toFile(), StandardCharsets.UTF_8);
+        Assert.assertTrue(log.contains("configuringAFileLoggerRotatingDaily - line1"));
+        Assert.assertTrue(log.contains("configuringAFileLoggerRotatingDaily - line2"));
+    }
+
+    /**
+     * A test that ensures that the log file rotates in the next day
+     */
+    @Test
+    public void logFileProperlyRotatingInNextDayWithDailyRotation() throws IOException {
+        ZonedDateTime zonedDateTime1 = ZonedDateTime.now();
+        ZonedDateTime zonedDateTime2 = zonedDateTime1.plusDays(1);
+        MockedStatic<ZonedDateTime> mockedLocalDateTime = Mockito.mockStatic(ZonedDateTime.class, Mockito.withSettings().defaultAnswer(Answers.CALLS_REAL_METHODS));
+
+        mockedLocalDateTime.when(ZonedDateTime::now).thenReturn(zonedDateTime1);
+        // Check that the log file is created with the corresponded format (Today date)
+        Path logFile = folder.getRoot().toPath().resolve("file");
+        LogFileAuditLogger logFileAuditLogger = new LogFileAuditLogger(logFile.toString(), 0, 2, null, true);
+        logFileAuditLogger.log("configuringAFileLoggerRotatingDaily - line1");
+        Path logFileRotating = folder.getRoot().toPath().resolve(logFileAuditLogger.computePattern());
+        Assert.assertTrue(logFileRotating.toFile().exists());
+
+        // Check that there is ONLY one file generated at this point (Today date)
+        String directoryPath = logFile.toFile().getParent();
+        Collection<File> directoryFiles = FileUtils.listFiles(new File(directoryPath), new RegexFileFilter(".*" + logFile.toFile().getName() + LogFileAuditLogger.DAILY_ROTATING_FILE_REGEX_PATTERN), DirectoryFileFilter.DIRECTORY);
+        Assert.assertEquals(directoryFiles.size(),1);
+
+        // Log something and check it appears in the logger file
+        String log = Util.loadFile(logFileRotating.toFile(), StandardCharsets.UTF_8);
+        Assert.assertTrue(log.contains("configuringAFileLoggerRotatingDaily - line1"));
+
+        // Increase +1 day
+        mockedLocalDateTime.when(ZonedDateTime::now).thenReturn(zonedDateTime2);
+
+        // Log something else
+        logFileAuditLogger.log("configuringAFileLoggerRotatingDaily - line2");
+
+        // Check that the corresponded is the ONLY one which appear on this file (Today +1)
+        logFileRotating = folder.getRoot().toPath().resolve(logFileAuditLogger.computePattern());
+        log = Util.loadFile(logFileRotating.toFile(), StandardCharsets.UTF_8);
+        Assert.assertTrue(log.contains("configuringAFileLoggerRotatingDaily - line2"));
+        Assert.assertFalse(log.contains("configuringAFileLoggerRotatingDaily - line1"));
+
+        mockedLocalDateTime.close();
```

Comment:
If the test ends before reaching this line (test failure), then this resource will not be closed. Since `MockedStatic` is an `Autocloseable` you can wrap everything in a try with resource block.